x-pack/plugin/core/src/main/java/org/elasticsearch/license/TransportGetTrialStatusAction.java

```diff
@@ -49,9 +53,12 @@ protected void masterOperation(
         ClusterState state,
         ActionListener<GetTrialStatusResponse> listener
     ) throws Exception {
-        LicensesMetadata licensesMetadata = state.metadata().custom(LicensesMetadata.TYPE);
-        listener.onResponse(new GetTrialStatusResponse(licensesMetadata == null || licensesMetadata.isEligibleForTrial()));
-
+        if (licenseService instanceof ClusterStateLicenseService) {
+            LicensesMetadata licensesMetadata = state.metadata().custom(LicensesMetadata.TYPE);
+            listener.onResponse(new GetTrialStatusResponse(licensesMetadata == null || licensesMetadata.isEligibleForTrial()));
```

Comment:
This almost feel like an interface method, i.e. `LicenseService#isEligibleForTrial`. I am aware that existing logic passes in a `LicenseMetadata` which makes abstraction harder. This is a similar situation as the above `TransportGetLicenseAction` which feels like another tech debt to address in future.